app/jobs/autoscaling.py

```diff
@@ -198,9 +208,133 @@ def set_desired_capacity(asg, compute_desired_instances, can_scale=True):
     if can_scale:
         if DEBUG:
             print cmd
+        if num_desired < previous_desired:
+            start_draining(asg, previous_desired - num_desired)
+        elif num_desired > previous_desired:
+            stop_draining(asg, num_desired - previous_desired)
         aws_command(cmd)
 
 
+def instances_in(asg):
+    return [item["InstanceId"] for item in asg["Instances"]]
+
+
+def iso_time_now():
+    return datetime.datetime.now().isoformat()
```

Comment:
could we please use unixtime instead?   the code to parse and unparse unixtime for tags already exists in this file and unixtime is used in the ASG tags, see for example the line above `tag_value = "{unixtime:d}::{value:d}".format(unixtime=int(time.time()), value=int(value))`

Reply:
if for debugging purposes you wish to convert from unixtime to human readable local time, the python function time.localtime(unixtime_integer) will do so.

Reply:
Yes. I was using iso time because that's how AWS gives you the LaunchTime. I will convert the AWS output to unixtime